test/hotspot/jtreg/runtime/jni/daemonDestroy/Main.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/**
+ * Simple application that checks that shutdown does not commence until
+ * after the last non-daemon thread has terminated. Reporting failure is
+ * tricky because we can't uses exceptions (as they are ignored from Shutdown
+ * hooks) and we can't call System.exit. So we rely on System.out being checked
+ * in the TestDaemonDestroy application.
+ */
+public class Main {
+
+    static volatile Thread t1;
+
+    // Note: not normal main(String[] args)
```

Comment:
I will delete the comment. It was really a note-to-self after I finally debugged the NoSuchMethodError I was getting because I'd copied code that looks for a regular "main".